### PR TITLE
Creat a docker machine to let the tester run on docker

### PR DESCRIPTION
**Pull request by fabiofabbrucci**

### CODE REVIEW

**Review by renie — CHANGES_REQUESTED** [review]
First of all, thanks for the initiative. It is good to have a container option to run the project.

That said, three things would be nice:

- Update README to let the user know how to use the docker solution.
- The `Dockerfile` just sets an image from and run update. What is kinda not necessary. We could keep use just `docker-compose` and remove `Dockerfile`.
To do that, you can just remove the `build` directive from `Dockerfile` and replace the value of `image` to the dockerhub tag directly.
- Instead of `node:slim`, I think we could use `node:lts-alpine3.17`. Some reasons to do that: 
   - A more specific tag is usually easier to debug and won't change so easily (or even at all)
   - `slim` tag has more OS dependencies, for being [based on Debian](https://github.com/nodejs/docker-node/blob/f996e97a2e6d2aae2de3b869e083a253733f07a8/19/bullseye-slim/Dockerfile), so it is a bit larger image. And because it has more deps, it will have some more security vulnerabilities. On the other hand, `alpine` images are [based on Alpine linux](https://github.com/nodejs/docker-node/blob/f996e97a2e6d2aae2de3b869e083a253733f07a8/18/alpine3.17/Dockerfile), so it has less deps.
   - It is based on the last LTS. The image `node:slim` is based on the last released version, so now it is 19.6, tomorrow might be 19.8...20.3... There could be changes on node and/or vite and react that does not work together anymore.

**Comment by fabiofabbrucci** [comment]
Thank you @renie for your kind feedback.
I've just pushed the fix you've just suggested.

I really appreciate your comments because they are extremely future-proof! 👏

File: README.md [hunk 1]
```diff
@@ -3,7 +3,7 @@ The only purpose of this repo is to be used as technical test for new QA Enginee
 
 
 ## What are the requirements to setup this project?
-This project requires `NodeJS` intalled on version 18+.
+This project requires `Docker Desktop` intalled.
```
Comment (renie) [inline README.md]: This project requires `NodeJS`, on version 18, or `Docker` installed. Depending on the way you prefer to run it.

File: README.md [hunk 2]
```diff
@@ -12,14 +12,28 @@ Just clone the repo and install deps.
...
+cd docker
+docker-compose up
+```
+
+### How to run using your local node
+
+Install dependencies in the new image.
+
+```
+cd qatest
```
Comment (renie) [inline README.md]: `cd qatest` is already on the previous step after cloning

File: README.md [hunk 3]
```diff
@@ -3,7 +3,7 @@ The only purpose of this repo is to be used as technical test for new QA Enginee
 
 
 ## What are the requirements to setup this project?
-This project requires `NodeJS` intalled on version 18+.
+This project requires `Docker Desktop` intalled.
 
 
 ## How to setup this project?
```
Comment (renie) [inline README.md:5]: It says "install deps", but we are not installing deps on this step anymore.

File: README.md [hunk 4]
```diff
@@ -12,14 +12,28 @@ Just clone the repo and install deps.
...
+### How to run using Docker
+
+```
+cd docker
+docker-compose up
+```
+
+### How to run using your local node
+
+Install dependencies in the new image.
```
Comment (renie) [inline README.md]: Install dependencies and run `dev` script.

File: README.md [hunk 5]
```diff
@@ -12,14 +12,28 @@ Just clone the repo and install deps.
...
 cd qatest
-npm i
 ```
 
 ## How to run this project?
-Run `dev` script from NPM.
 
-```bash
+### How to run using Docker
+
```
Comment (renie) [inline README.md:16]: Go to `docker` folder and start docker container.

**Review by renie — CHANGES_REQUESTED** [review]
Now there are just some minor fix to align docs with the real process.
The rest is pretty good, I've just tested with and without docker on my machine and everything works like a charm.
Thank you!

**Review by renie — APPROVED** [review]

**Pull request merged**